scripts/xtensa-build-zephyr.py

```diff
@@ -477,6 +477,26 @@ def get_build_and_sof_version(abs_build_dir):
 
 	return sof_fw_version, sof_build_version
 
+def rmtree_if_exists(directory):
+	"This is different from ignore_errors=False because it deletes everything or nothing"
+	if os.path.exists(directory):
+		shutil.rmtree(directory)
+
+def clean_staging(platform):
+	print(f"Cleaning {platform} from {STAGING_DIR}")
+
+	rmtree_if_exists(STAGING_DIR / "sof-info" / platform)
+
+	sof_output_dir = STAGING_DIR / "sof"
+
+	# --use-platform-subdir
+	rmtree_if_exists(sof_output_dir / platform)
+
+	# Remaining .ri and .ldc files
+	for f in sof_output_dir.glob(f"**/sof-{platform}.*"):
+		os.remove(f)
```

Comment:
Wouldn't this delete all platforms, even those not build in this run? Or that is the intention?

Reply:
It starts by cleaning only the platforms _requested_ in this run, see the `{platform}` argument.